app/app.go

```diff
@@ -602,10 +616,101 @@ func (app *RegenApp) BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock) a
 // EndBlocker application updates every end block
 func (app *RegenApp) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.ResponseEndBlock {
 	resp := app.mm.EndBlock(ctx, req)
+
+	// TODO: update with mainnet validatorset
+	testnetPubKeys, err := GetTestnetValidatorSetPubKeys()
+	if err != nil {
+		panic(err)
+	}
+
+	// before the patch height continue the current wrong behavior.
+
+	// at patch height
+	// for all validators in app state (app.StakingKeeper.GetAllValidators), create a ValidatorUpdate with the correct power
+	// for all remaining validators in []VoteInfo that are not in app state, set their voting power to zero
+	if ctx.BlockHeight() == PATCH_HEIGHT {
+		var updates []abci.ValidatorUpdate
+		validators := app.StakingKeeper.GetAllValidators(ctx)
+		inValidatorSet := map[string]bool{}
+
+		for _, validator := range validators {
+			tmProtoPk, err := validator.TmConsPublicKey()
+			if err != nil {
+				panic(err)
+			}
+
+			valCons, err := validator.GetConsAddr()
+			if err != nil {
+				panic(err)
+			}
+
+			if validator.Jailed {
+				for _, voteInfo := range app.votesInfo {
```

Comment:
@aaronc returning validator which does not exist in tendermint state is halting the network with this error
```
Sep 15 09:35:24 regen-devnet-val-01 cosmovisor[19623]: 9:35AM ERR failed to apply block err="commit failed for application: error changing validator set: failed to find validator 001F90AE99B52F548845DA4D93755C3D4B829097 to remove" height=600 module=consensus
```